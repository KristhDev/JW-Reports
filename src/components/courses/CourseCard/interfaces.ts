import { Course } from '../../../interfaces/courses';

export interface CourseCardProps {
    course: Course;
    onActiveOrSuspend: () => void;
    onDelete: () => void;
    onFinishOrStart: () => void;
}