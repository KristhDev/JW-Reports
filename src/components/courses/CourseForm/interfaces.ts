export interface CourseFormValues {
    person_name: string;
    person_about: string;
    person_address: string;
    publication: string;
}