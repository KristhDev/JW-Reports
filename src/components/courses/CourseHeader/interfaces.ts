export interface RevisitHeaderProps {
    editButton: boolean;
    deleteButton: boolean;
}