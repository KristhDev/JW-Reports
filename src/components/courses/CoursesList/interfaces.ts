import { CourseFilter } from '../../../interfaces/courses';

export interface CoursesListProps {
    emptyMessage: string;
    filter: CourseFilter;
    title: string;
}

export interface ListEmptyComponentProps {
    msg: string;
}