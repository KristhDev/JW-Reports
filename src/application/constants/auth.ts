import { ItemOption, Precursor } from '@infrasturcture/interfaces';

/**
 * The precursors options to select
 */
export const PRECURSORS_OPTIONS: ItemOption[] = [
    { label: 'Ninguno', value: 'ninguno' },
    { label: 'Auxiliar', value: 'auxiliar' },
    { label: 'Regular', value: 'regular' },
    { label: 'Especial', value: 'especial' }
];

export const precursors: Record<Uppercase<Precursor>, Precursor> = {
    AUXILIAR: 'auxiliar',
    ESPECIAL: 'especial',
    NINGUNO: 'ninguno',
    REGULAR: 'regular'
}

export const authMessages = {
    CONFIRM_PASSWORD_EMPTY: 'La confirmación de la contraseña es requerida.',
    EMAIL_ALREADY_REGISTERED: 'Ya existe un usuario con este correo.',
    EMAIL_EMPTY: 'El correo no puede estar vacío.',
    EMAIL_INVALID: 'Correo electrónico inválido.',
    EMAIL_UPDATE_UNCHANGED: 'Para actualizar tu correo debes cambiarlo.',
    NAME_EMPTY: 'El nombre no puede estar vacío.',
    NAME_MIN_LENGTH: 'El nombre debe tener al menos 2 caracteres.',
    PASSWORD_EMPTY: 'La contraseña no puede estar vacía.',
    PASSWORD_MIN_LENGTH: 'La contraseña debe tener al menos 6 caracteres.',
    PASSWORD_NOT_MATCH: 'Las contraseñas no coinciden.',
    PASSWORD_UPDATED: 'Ha actualizado su contraseña correctamente.',
    PROFILE_UPDATED: 'Ha actualizado su perfil correctamente.',
    SURNAME_EMPTY: 'Los apellidos no pueden estar vacíos.',
    SURNAME_MIN_LENGTH: 'Los apellidos deben tener al menos 2 caracteres.',
    UNATHENTICATED: 'Para realizar está acción debe iniciar sesión.',
    UNAUTHORIZED: 'No tiene permiso para realizar está acción.',
}

export const precursorMessages = {
    PRECURSOR_EMPTY: 'El campo precursor es requerido.',
    PRECURSOR_INVALID: 'Por favor seleccione una opción de precursor.',
}