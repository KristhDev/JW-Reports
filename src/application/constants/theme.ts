import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];