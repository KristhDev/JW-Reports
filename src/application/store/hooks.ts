import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();