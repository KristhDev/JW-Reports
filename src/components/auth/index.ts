export { CredentialsForm } from './CredentialsForm';
export { ForgotPasswordForm } from './ForgotPasswordForm';
export { LoginForm } from './LoginForm';
export { ProfileForm } from './ProfileForm';
export { RegisterForm } from './RegisterForm';