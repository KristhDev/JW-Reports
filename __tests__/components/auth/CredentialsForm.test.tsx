import React from 'react';
import { act, fireEvent, render, screen, waitFor } from '@testing-library/react-native';

import { CredentialsForm } from '../../../src/components/auth';

import { useAuth, useStatus, useTheme } from '../../../src/hooks';

import { User } from '../../../src/interfaces/auth';

import { darkColors } from '../../../src/theme';

const testUser: User = {
    id: '05ef0d0c-0f7a-4512-b705-6da279d88503',
    name: 'Celestino',
    surname: 'Wilderman',
    email: 'Ernestine_Doyle@yahoo.com',
    precursor: 'ninguno',
    createdAt: '2021-03-10T12:00:00.000Z',
    updatedAt: '2021-03-10T12:00:00.000Z',
}

const updateEmailMock = jest.fn();
const updatePasswordMock = jest.fn().mockImplementation(() => Promise.resolve());
const setErrorFormMock = jest.fn();

jest.mock('../../../src/hooks/useAuth.ts');
jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <CredentialsForm /> component', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    (useAuth as jest.Mock).mockReturnValue({
        state: { isAuthLoading: false, user: testUser },
        updateEmail: updateEmailMock,
        updatePassword: updatePasswordMock
    });

    (useStatus as jest.Mock).mockReturnValue({
        setErrorForm: setErrorFormMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors }
    });

    it('should to match snapshot', async () => {
        await waitFor(() => {
            render(<CredentialsForm />);
        });

        await act(() => {
            expect(screen.toJSON()).toMatchSnapshot();
        });
    });

    it('should render user.email in form', async () => {
        await waitFor(() => {
            render(<CredentialsForm />);
        });

        await act(async() => {
            await waitFor(() => {
                const inputsText = screen.getAllByTestId('form-field-text-input');
                expect(inputsText[0].props.value).toBe(testUser.email);
            });
        });
    });

    it('should call setErrorForm then email isnt change', async () => {
        await waitFor(() => {
            render(<CredentialsForm />);
        });

        await act(async() => {
            await waitFor(() => {
                const touchables = screen.getAllByTestId('button-touchable');
                fireEvent.press(touchables[0]);

                expect(setErrorFormMock).toHaveBeenCalledTimes(1);
            });
        });
    });

    it('should call updateEmail then email form is valid', async () => {
        await waitFor(() => {
            render(<CredentialsForm />);
        });

        const newEmail = 'tester@gmail.com';

        await act(async() => {
            await waitFor(() => {
                const inputsText = screen.getAllByTestId('form-field-text-input');
                fireEvent(inputsText[0], 'onChangeText', newEmail);

                const touchables = screen.getAllByTestId('button-touchable');
                fireEvent.press(touchables[0]);

                expect(updateEmailMock).toHaveBeenCalledTimes(1);
            });

            expect(updateEmailMock).toHaveBeenCalledWith({ email: newEmail }, expect.any(Function));
        });
    });

    it('should disabled button when isAuthLoading and loadingEmail is true', async () => {
        (useAuth as jest.Mock).mockReturnValue({
            state: { isAuthLoading: true, user: testUser },
            updateEmail: updateEmailMock,
            updatePassword: updatePasswordMock
        });

        await waitFor(() => {
            render(<CredentialsForm />);
        });

        const newEmail = 'tester@gmail.com';

        await act(async() => {
            await waitFor(() => {
                const inputsText = screen.getAllByTestId('form-field-text-input');
                fireEvent(inputsText[0], 'onChangeText', newEmail);

                const touchables = screen.getAllByTestId('button-touchable');
                fireEvent.press(touchables[0]);

                expect(touchables[0].props.accessibilityState.disabled).toBeTruthy();
            });
        });
    });

    it('should call setErrorForm when password form is empty or invalid', async () => {
        await waitFor(() => {
            render(<CredentialsForm />);
        });

        await act(async() => {
            await waitFor(() => {
                const touchables = screen.getAllByTestId('button-touchable');
                fireEvent.press(touchables[1]);

                expect(setErrorFormMock).toHaveBeenCalledTimes(1);
            });
        });
    });

    it('should call updatePassword when password form is valid', async () => {
        await waitFor(() => {
            render(<CredentialsForm />);
        });

        const newPass = 'new-password-1234';

        await act(async() => {
            await waitFor(() => {
                const inputsText = screen.getAllByTestId('form-field-text-input');
                fireEvent(inputsText[1], 'onChangeText', newPass);
                fireEvent(inputsText[2], 'onChangeText', newPass);

                const touchables = screen.getAllByTestId('button-touchable');
                fireEvent.press(touchables[1]);

                expect(updatePasswordMock).toHaveBeenCalledTimes(1);
            });

            expect(updatePasswordMock).toHaveBeenCalledWith({ password: newPass }, expect.any(Function));
        });
    });

    it('should disabled button when isAuthLoading and loadingPassword is true', async () => {
        (useAuth as jest.Mock).mockReturnValue({
            state: { isAuthLoading: true, user: testUser },
            updateEmail: updateEmailMock,
            updatePassword: updatePasswordMock
        });

        await waitFor(() => {
            render(<CredentialsForm />);
        });

        const newPass = 'new-password-1234';

        await act(async() => {
            await waitFor(() => {
                const inputsText = screen.getAllByTestId('form-field-text-input');
                fireEvent(inputsText[1], 'onChangeText', newPass);
                fireEvent(inputsText[2], 'onChangeText', newPass);

                const touchables = screen.getAllByTestId('button-touchable');
                fireEvent.press(touchables[1]);

                expect(touchables[1].props.accessibilityState.disabled).toBeTruthy();
            });
        });
    });
});