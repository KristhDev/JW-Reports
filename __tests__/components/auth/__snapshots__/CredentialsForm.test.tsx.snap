// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CredentialsForm /> component should to match snapshot 1`] = `
<View
  style={
    Object {
      "paddingVertical": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "flex-start",
        "paddingBottom": 30,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-field-label"
      >
        Correo:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              cursorColor="#C0A7E1"
              keyboardType="email-address"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSelectionChange={[Function]}
              placeholder="Ingrese su correo"
              placeholderTextColor="#BFBFBF"
              selection={
                Object {
                  "end": 25,
                  "start": 25,
                }
              }
              selectionColor="#A0B9E2"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                    "paddingRight": 5,
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
              testID="form-field-text-input"
              value="Ernestine_Doyle@yahoo.com"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#BFBFBF",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Array [
            Object {
              "backgroundColor": "#C0A7E1",
              "borderRadius": 10,
            },
            Object {
              "marginTop": 30,
              "paddingHorizontal": 20,
            },
          ],
        ]
      }
      testID="button-touchable"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 20,
              "paddingVertical": 8,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "color": "#121212",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 2,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
          testID="button-text"
        >
          Cambiar correo
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "flex-start",
        "paddingBottom": 30,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-field-label"
      >
        Nueva contraseña:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              cursorColor="#C0A7E1"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSelectionChange={[Function]}
              placeholder="Ingrese su contraseña"
              placeholderTextColor="#BFBFBF"
              secureTextEntry={true}
              selection={
                Object {
                  "end": 0,
                  "start": 0,
                }
              }
              selectionColor="#A0B9E2"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                    "paddingRight": 5,
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
              testID="form-field-text-input"
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="eye-btn-touchable"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#BFBFBF",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-field-label"
      >
        Confirmar contraseña:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              cursorColor="#C0A7E1"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSelectionChange={[Function]}
              placeholder="Confirme su contraseña"
              placeholderTextColor="#BFBFBF"
              secureTextEntry={true}
              selection={
                Object {
                  "end": 0,
                  "start": 0,
                }
              }
              selectionColor="#A0B9E2"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                    "paddingRight": 5,
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
              testID="form-field-text-input"
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="eye-btn-touchable"
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#BFBFBF",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Array [
            Object {
              "backgroundColor": "#C0A7E1",
              "borderRadius": 10,
            },
            Object {
              "marginTop": 30,
              "paddingHorizontal": 20,
            },
          ],
        ]
      }
      testID="button-touchable"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 20,
              "paddingVertical": 8,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "color": "#121212",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 2,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
          testID="button-text"
        >
          Cambiar contraseña
        </Text>
      </View>
    </View>
  </View>
</View>
`;
